cmd/server/main.go

```diff
@@ -39,80 +38,64 @@ type Arguments struct {
 
 func mainWithArgs(ctx context.Context, args []string, logger golog.Logger) error {
 	var argsParsed Arguments
+
 	if err := utils.ParseFlags(args, &argsParsed); err != nil {
 		return err
 	}
+
 	if argsParsed.Port == 0 {
+		golog.Global.Debugf("using default port %d ", defaultPort)
 		argsParsed.Port = utils.NetPortFlag(defaultPort)
 	}
 
-	lidarComponents := search.Devices()
-	if len(lidarComponents) != 0 {
-		golog.Global.Debugf("detected %d lidar devices", len(lidarComponents))
-		for _, comp := range lidarComponents {
+	// Check if USB device is available
+	// TODO(jeremy): add search filter for product and model info to confirm rplidar is present instead of assuming
+	usbDevices := usb.Search(
+		usb.SearchFilter{},
+		func(vendorID, productID int) bool {
+			return true
+		})
+
+	if len(usbDevices) != 0 {
+		golog.Global.Debugf("detected %d lidar devices", len(usbDevices))
+		for _, comp := range usbDevices {
 			golog.Global.Debug(comp)
```

Comment:
As before, these loggers instances should use the logger that was passed to the function.

Reply:
Done.